client/dev/navbar.jsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+
+class NavLink extends React.Component{
+  render(){
+    return (
+      <a href={this.props.url}>{this.props.linktext}</a>
+    )
+  }
+}
+
+class NavItem extends React.Component {
+  makeLink(){
+    return <NavLink url={this.props.url} linktext={this.props.linktext} />;
+  }
+  
+  makeSubMenu(){
+    return <NavBar items={this.props.submenu} />;
+  }
+  
+  makeNavContent(){
+    var content = [this.makeLink()];
+    if(this.props.submenu){
+      content.push(this.makeSubMenu());
+    }
+    return content;
+  }
+  
+  render(){
+    var content = this.makeNavContent();
+    return (
+      <li>
+        {content}
+      </li>
+    );
+  }
+}
+
+class NavBar extends React.Component {
+  makeItem(item){
+    return ( <NavItem linktext={item.linktext} url={item.url} submenu={item.submenu} /> )
+  }
+  
+  render(){
+    var items = this.props.items.map(this.makeItem);
```

Comment:
Use `const` instead of `var` because ES6. Your tools here are `const` and `let` and they are generally more useful than `var`. `const` is a constant and can't be modified after assignment (although this just means for objects that it must point to the same object reference and the object it represents can change). `let` can change after assignment. They key difference with `const` and `let` from `var` is that `const` and `let` are able to be defined in block scope unlike `var`. For example, if you define a const inside an if statement it will only be available in that scope, where if you were to define a var inside an if statement it would be available to everything in that function.

Prefer this pattern instead:

```javascript
const NavBar = props => {
  return (
    <ul>
      { props.items.map( item, i => {
        return ( <NavItem key={i} linktext={item.linktext} url={item.url} submenu={item.submenu} );
       }}
    </ul>
```
Two call-outs:
1. No need for separate function to make each item, but if you do want to do it you still can. This way you're not caching the result of the map by creating another variable though.
2. Each item in iterated component should have a key prop so that React can keep track of which is which.